### PR TITLE
fixing multiple issues

### LINKED ISSUE #21: [BUG] Codeowner files with Giltlab nested teams not parsed properly
Example of codeowner file:
 
`* @group-x/subgroup-y/subgroup-z`

throws `invalid owner format '@group-x/subgroup-y/subgroup-z' at position 3 `

Gitlab docs:
https://docs.gitlab.com/ee/user/project/codeowners/#add-a-group-as-a-code-owner

### PR DESCRIPTION
**Pull request by rwese**
fixes #21, #20, #19, #24, #25

### CODE REVIEW

File: parse.go [hunk 1]
```diff
@@ -187,3 +425,30 @@ func isOwnersChar(ch rune) bool {
 	}
 	return isAlphanumeric(ch)
 }
+
+// isSectionChar matches characters that are allowed in owner definitions
```
Comment (rwese) [inline parse.go:429]:
```suggestion
// isSectionChar matches characters that are allowed for section names
```

File: parse.go [hunk 2]
```diff
@@ -187,3 +425,30 @@ func isOwnersChar(ch rune) bool {
...
+func isSectionChar(ch rune) bool {
+	switch ch {
+	case '.', '@', '/', '_', '%', '+', '-', ' ':
+		return true
+	}
+	return isAlphanumeric(ch)
+}
+
+// isSectionEnd matches characters that are allowed in section definitions
```
Comment (rwese) [inline parse.go:438]:
```suggestion
// isSectionEnd matches characters ends each section block
// e.g. [Section Name][<approval count>]
```

File: parse.go [hunk 3]
```diff
@@ -187,3 +425,30 @@ func isOwnersChar(ch rune) bool {
...
+func isSectionEnd(ch rune) bool {
+	switch ch {
+	case ']':
+		return true
+	}
+	return false
+}
+
+// isSectionStart matches characters that are allowed in section definitions
```
Comment (rwese) [inline parse.go:447]:
```suggestion
// isSectionStart defines characters starting the beginning of a section
// - `^` starts an optional section
```

**Comment by hmarr** [comment]
Hi @rwese, thanks for submitting this. I've (separately) merged several of the PRs that you'd merged into this one, but it looks like this also contains some other useful changes — in particular, adding optional support for GitLab sections via `WithSectionSupport`. If you'd be interested in submitting a PR with just that change I'd happily merge it.

**Comment by rwese** [comment]
Will do, thanks!

**Pull request closed without merging**